app/models/mixins/miq_provision_quota_mixin.rb

```diff
@@ -275,25 +275,13 @@ def quota_find_active_prov_request_by_tenant(options)
   end
 
   def quota_find_active_prov_request(_options)
-    prov_req_ids = MiqQueue.where(
-      :class_name  => %w(MiqProvisionRequest ServiceTemplateProvisionRequest),
-      :method_name => 'create_request_tasks',
-      :state       => 'dequeue'
-    ).pluck(:instance_id)
-
-    prov_ids = []
-    MiqQueue
-      .where(:method_name => 'deliver', :state => %w(ready dequeue), :class_name => 'MiqAeEngine')
-      .where("tracking_label like ?", '%_provision_%')
-      .each do |q|
-        if q.args
-          args = q.args.first
-          prov_ids << args[:object_id] if args[:object_type].include?('Provision') && !args[:object_id].blank?
-        end
-      end
-    prov_req_ids += MiqRequestTask.where(:id => prov_ids).pluck("miq_request_id")
-
-    MiqRequest.where(:id => prov_req_ids.compact.uniq)
+    MiqRequest.where(
+      :approval_state => 'approved',
+      :type           => %w(MiqProvisionRequest ServiceTemplateProvisionRequest),
+      :request_state  => %w(active queued pending),
+      :status         => 'Ok',
+      :process        => true
+    ).where.not(:id => id)
```

Comment:
@tinaafitz This is a much clearer solution and almost completely gets the quota logic away from doing lookups in `MiqQueue`. (One call left in this file)

My one concern is the possibility for false-positive MiqRequest instances being returned.  We had that issue before, but it is slightly higher now because in the past clearing a queue would have removed the request from being returned from this method.  Now clearing the queue or some other error could leave the request in what looks like a valid state.

Wondering if it makes sense to log a count and list of IDs that are being returned here.  Or maybe we create a small script in the `tools` directory to help users identify what MiqRequest objects are in play for quota.

Thoughts?

Reply:
I agree that this is a much clearer solution than the previous MiqQueue lookup logic. Your false-positive concern is valid. While I was testing this changes, I did exactly that and added logging for the count and request ID's being included in the active counts. I'm in favor of the tools/script approach to help users identify the MiqRequest objects in play for quota.

Reply:
A major difference in this approach is that requests will be more likely to be denied based on the active requests, whereas before, the requests would be much more likely to be approved and cause quota to go negative.

Reply:
Sound good.  Please open a git issue for creating that tools script with a reference back to this PR.  I think that is a task we can assign to @billfitzgerald0120.